Sources/SourceKitLSP/Swift/CodeActions/ConvertStringConcatenationToStringInterpolation.swift

```diff
@@ -0,0 +1,138 @@
+//
+
+import LanguageServerProtocol
+import SwiftRefactor
+import SwiftSyntax
+
+struct ConvertStringConcatenationToStringInterpolation: SyntaxRefactoringProvider {
+  static func refactor(syntax: ExprListSyntax, in context: Void) -> ExprListSyntax? {
+    guard let (componentsOnly, commonPounds) = syntax.preflight() else {
+      return nil
+    }
+    
+    var ret: StringLiteralSegmentListSyntax = []
+    for component in componentsOnly {
+      if var stringLiteral = StringLiteralExprSyntax(component) {
+        stringLiteral.pounds = commonPounds
+        ret += stringLiteral.segments
+      } else {
+        ret.append(.expressionSegment(ExpressionSegmentSyntax(
+          pounds: commonPounds,
+          expressions: [
+            LabeledExprSyntax(expression: component.trimmed)
+          ]
+        )))
+      }
+    }
+    
+    return [
+      ExprSyntax(StringLiteralExprSyntax(
+        openingPounds: commonPounds,
+        openingQuote: "\"",
+        segments: ret,
+        closingQuote: "\"",
+        closingPounds: commonPounds
+      ))
+    ]
+  }
+}
+
+extension ConvertStringConcatenationToStringInterpolation: SyntaxRefactoringCodeActionProvider {
+  static let title: String = "Convert String Concatenation to String Interpolation"
+  
+  static func nodeToRefactor(in scope: SyntaxCodeActionScope) -> ExprListSyntax? {
+    guard let token = scope.innermostNodeContainingRange,
+    let exprList = token.findParentOfSelf(
+      ofType: ExprListSyntax.self,
+      stoppingIf: {
+        $0.kind == .codeBlockItem || $0.kind == .memberBlockItem
+      }
+    ) else {
+      return nil
+    }
+    
+    return exprList
+  }
+}
+
+fileprivate extension ExprListSyntax {
+  func preflight() -> (componentsOnly: [ExprListSyntax.Element], commonPounds: TokenSyntax?)? {
+    var iter = makeIterator()
+    guard let first = iter.next() else {
+      return nil
+    }
+    
+    var hasStringLiterals = false
+    var longestPoundsText: String?
+    var componentsOnly = [first]
+
+    if let stringLiteral = StringLiteralExprSyntax(first) {
+      hasStringLiterals = true
+      longestPoundsText = stringLiteral.poundsText
+    }
+    
+    while let plus = iter.next(), let stringComponent = iter.next() {
+      guard let plus = BinaryOperatorExprSyntax(plus), case .binaryOperator("+") = plus.operator.tokenKind else {
+        return nil
+      }
+      
+      if let stringLiteral = StringLiteralExprSyntax(stringComponent) {
+        hasStringLiterals = true
+        if let openingPoundsText = stringLiteral.poundsText, openingPoundsText.count > (longestPoundsText?.count ?? 0) {
+          longestPoundsText = openingPoundsText
+        }
+      }
+      
+      componentsOnly.append(stringComponent)
+    }
+    
+    guard hasStringLiterals else {
+      return nil
+    }
+    
+    let commonPounds: TokenSyntax? = if let longestPoundsText {
+      .rawStringPoundDelimiter(longestPoundsText)
+    } else {
+      nil
+    }
```

Comment:
Could we avoid this if `longestPoundsText` was of type `TokenSyntax?` from the start?